fix: add mapping of advertiser_tracking_enabled and enable extInfo with empty string

ItsSudip opened the pull request:
## What are the changes introduced in this PR?

Fix the logic of populating extInfo, setting an empty string for null values, apart from dropping the entire extInfo and use advertiser_tracking_enabled.

## What is the related Linear task?

Resolves INT-3910

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

devops-github-rudderstack commented:
Allure Test reports for this run are available at:
- Allure Report: [View Report](https://rudderlabs.github.io/test-reports/rudder-transformer/4491/index.html)

krishna2020 commented on src/v0/destinations/snapchat_conversion/utilsV3.js:
```diff
@@ -128,8 +128,7 @@ const getExtInfo = (message) => {
   ];
-
-  return extInfo.some((value) => value == null) ? null : extInfo;
+  return extInfo.map((value) => (value == null ? '' : value));
```
I think the expectation will be to not send a value when it's not present.
So you also need to use filter for dropping the fields that are null. There's difference between the field being empty vs null vs 'null' vs non-existent field

ItsSudip replied:
https://developers.snap.com/api/marketing-api/Conversions-API/Parameters As per the doc, I don't see sending an empty string, but the customer suggested sending empty strings.

krishna2020 replied:
Extinfo requires all values and needs them to be in the order indexed below in the extinfo object specs.
It's assigning specific field [based on index](https://developers.snap.com/api/marketing-api/Conversions-API/Parameters#extinfo) - that's the reason for customer asking to send empty values to maintain the order. Then this change looks good

ItsSudip commented:
> I don't see any usage of `advertiser_tracking_enabled` though PR title says it

Actually, it is there in the mapping, but the populated value is not being used. So I added the use of it through commonProperties.app_data. I have updated one test case regarding this.
